main.go

```diff
@@ -70,13 +70,15 @@ func printVersion() {
 
 // nolint:maintidx
 func main() {
-	var enableLeaderElection, version bool
+	var enableLeaderElection, enableSecretController, version bool
 
 	var metricsAddr, namespace, configurationName string
 
 	var goFlagSet goflag.FlagSet
 
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	flag.BoolVar(&enableSecretController, "enable-secret-controller", true,
```

Comment:
Do we need this flag?

If we're missing the Secret since non-existing, the Pod will not be started until those are available for the mount.

Reply:
we need to disable build-in reconciler to be able to generate secret using cert-manager

Reply:
I don't have any doubts about it, pretty sure it works. I'm just wondering why we should disable the secret reconciler since the CA and TLS would be created by `cert-manager`.

Capsule would just check if the Secret CA and TLS are there, and read their value. Pretty sure I'm missing something.

Reply:
due to this https://github.com/clastix/capsule/blob/master/controllers/secret/ca.go#L183. The controller will find a secret, but it will try to update it with its own generated data, which will lead to 
```
2022/05/06 11:50:35 http: TLS handshake error from 10.244.0.1:1547: remote error: tls: bad certificate
```
errors

Reply:
and there is one more problem. I recalled that we need to inject cert-manager annotation to tenant crd, as we also have a conversion webhook there. And i really don't know how to do it, as crds are not templated

Reply:
@MaxFedotov can't we use the `cert-manager`'s [CA Injector](https://cert-manager.io/docs/concepts/ca-injector/)?

Reply:
We can, but we need to template the name of the certificate in cert-manager annotation '
cert-manager.io/inject-ca-from: {{ .Release.Namespace }}/{{ include "capsule.fullname" . }}-webhook-cert'

And templates and variables can't be used in crds :(

Reply:
@MaxFedotov WDYT on moving CRD definition from `crds` folder to `templates`? Do you see any drawbacks to this?

We're already minor bumping upon any new CRD update, since we're informing users I don't see any issue on this.

Reply:
The only problem will be that resources in crds folder are not deleted with helm uninstall. If we move them to templates folder, they will be removed (and with the removing of crds all tenants and namespaces will be deleted as well). And one more thing to check is the ordering when applying resources. We have a  capsuleConfiguration crd and capsuleConfiguration resource with default configuration, so we need to ensure that crd will be applied first